sources/ghostdog/src/infiniband.rs

```diff
@@ -0,0 +1,197 @@
+use crate::error;
+use crate::error::Result;
+use snafu::{ensure, ResultExt};
+use std::ffi::OsString;
+use std::ops::BitAnd;
+use std::path::Path;
+use std::str::FromStr;
+use std::{fs, str};
+
+const SYS_CLASS_DIR: &str = "/sys/class";
+const SW_MNG_CLASS: &str = "infiniband";
+const SW_MNG_VALUE: &str = "SW_MNG";
+
+/// Generic device struct found from sysfs
+#[derive(Clone, Debug)]
+pub(crate) struct SysDevice {
+    pub(crate) name: OsString,
+}
+
+/// Finds all the devices in sysfs for a given class sorted by name
+pub(crate) fn find_devices_by_class(class: String) -> Result<Vec<SysDevice>> {
+    let devices_path = Path::new(SYS_CLASS_DIR).join(class);
+    let mut found_devices: Vec<SysDevice> = vec![];
+    if !devices_path.exists() {
+        // Nothing to do since no devices detected
+        return Ok(found_devices);
+    }
+
+    let sys_devices_dirs = fs::read_dir(devices_path).context(error::InfinibandSysDevicesSnafu)?;
+    for device in sys_devices_dirs {
+        let device = device.context(error::InfinibandDeviceSnafu)?;
+        let device_name: OsString = device.file_name();
+        found_devices.push(SysDevice { name: device_name })
+    }
+
+    found_devices.sort_by_key(|device| device.name.clone());
```

Comment:
With the derived impls:
```suggestion
    found_devices.sort();
```